From 7821b6a1baefd97757142d9c3227b5219e9f4568 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 24 Feb 2025 15:15:02 +0000
Subject: [PATCH] media: hevc_dec: Drop the new image formats until we have
 Mesa 24

Supporting GL rendering of the new HEVC decoder pixel formats requires
Mesa 24.2.5 or later. There are a couple of minor issues holding up
switching to Mesa 24.

Drop the new pixel formats from enum_fmt so that FFMpeg will use
the older ones that earlier versions of Mesa do support.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/platform/raspberrypi/hevc_dec/hevc_d_video.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/drivers/media/platform/raspberrypi/hevc_dec/hevc_d_video.c
+++ b/drivers/media/platform/raspberrypi/hevc_dec/hevc_d_video.c
@@ -284,8 +284,8 @@ static u32 pixelformat_from_sps(const st
 				const int index)
 {
 	static const u32 all_formats[] = {
-		V4L2_PIX_FMT_NV12MT_COL128,
-		V4L2_PIX_FMT_NV12MT_10_COL128,
+		//V4L2_PIX_FMT_NV12MT_COL128,
+		//V4L2_PIX_FMT_NV12MT_10_COL128,
 		V4L2_PIX_FMT_NV12_COL128,
 		V4L2_PIX_FMT_NV12_10_COL128,
 	};
@@ -298,11 +298,12 @@ static u32 pixelformat_from_sps(const st
 			pf = all_formats[index];
 	} else {
 		if (index == 0) {
-			if (sps->bit_depth_luma_minus8 == 0)
+/*			if (sps->bit_depth_luma_minus8 == 0)
 				pf = V4L2_PIX_FMT_NV12MT_COL128;
 			else if (sps->bit_depth_luma_minus8 == 2)
 				pf = V4L2_PIX_FMT_NV12MT_10_COL128;
 		} else if (index == 1) {
+			*/
 			if (sps->bit_depth_luma_minus8 == 0)
 				pf = V4L2_PIX_FMT_NV12_COL128;
 			else if (sps->bit_depth_luma_minus8 == 2)
